lib/numbers/matrix.js

```diff
@@ -1288,3 +1289,27 @@ matrix.isStrictlyColumnDD = function (M) {
   }
   return true;
 };
+
+/**
+ * Applies a function to every element of a vector or matrix (i.e. map).
+ * The function must take only one parameter.
+ *
+ * @param {Array} matrix
+ * @param {Function} function to apply to each element
+ * @return {Array} matrix operated on
+ */
+matrix.map = function (M, f) {
+  // M is n-by-m (n rows, m columns)
+  var n = M.length,
+      m = M[0].length,
+      i, j;
+
+  var res = matrix.deepCopy(M);
+
+  for (i=0; i<n; i++) {
+    for (j=0; j<m; j++) {
+      res[i][j] = f(M[i][j]);
```

Comment:
I think `.apply` would be better instead of calling the function but it's not that big of a deal.